.github/workflows/smoke_tests.yaml

```diff
@@ -1,45 +1,47 @@
-# only has to pass for python 3.9
 name: Smoke Tests
 
 on:
   push:
     branches:
-      main
+      - main
   pull_request:
     branches:
-      main
+      - main
 
 jobs:
   test:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
-      - name: Set up Python 3.9
+
+      - name: Set up Python 3.10
         uses: actions/setup-python@v5
         with:
-          python-version: 3.9
-      # Display the Python version being used
+          python-version: "3.10"
+
       - name: Display Python version
         run: python -c "import sys; print(sys.version)"
+
       - name: Set up file descriptor limit
-        run: |
-          ulimit -n 4096
+        run: ulimit -n 4096
+
       - name: Install and configure Poetry
         uses: snok/install-poetry@v1
         with:
           virtualenvs-create: true
           virtualenvs-in-project: true
+
       - name: Set up cache
         uses: actions/cache@v4
         id: cached-poetry-dependencies
         with:
           path: .venv
           key: venv-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('**/poetry.lock') }}
+
       - name: Install dependencies
-        run: poetry install --with "dev, test"
         if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install --with "dev, test"
```

Comment:
Maybe a dumb question, but does this `run:...` need to be indented to be "inside" the if?

Reply:
I thought that as well initially since I am not super familiar with the syntax of the workflow yaml files. But looking into it, it actually should be at the same level of indentation. The run line will be executed iff the if statement above it (at the same level of indentation) is true

Reply:
Interesting. Good to know as well!